subsys/bluetooth/host/l2cap.c

```diff
@@ -1167,27 +1175,34 @@ static void le_ecred_conn_req(struct bt_l2cap *l2cap, uint8_t ident,
 	}
 
 response:
-	buf = l2cap_create_le_sig_pdu(buf, BT_L2CAP_ECRED_CONN_RSP, ident,
-				      sizeof(*rsp) + (sizeof(scid) * i));
+	if (result != BT_L2CAP_LE_SUCCESS) {
+		buf = l2cap_create_le_sig_pdu(buf, BT_L2CAP_ECRED_CONN_RSP, ident,
+				      sizeof(*rsp) + (sizeof(scid)));
+		rsp = net_buf_add(buf, sizeof(*rsp));
 
-	rsp = net_buf_add(buf, sizeof(*rsp));
-	(void)memset(rsp, 0, sizeof(*rsp));
+		(void)memset(rsp, 0, sizeof(*rsp));
+		memset(dcid, 0, sizeof(scid));
 
-	if (result == BT_L2CAP_LE_ERR_UNACCEPT_PARAMS ||
-	    result == BT_L2CAP_LE_ERR_PSM_NOT_SUPP ||
-	    result == BT_L2CAP_LE_ERR_AUTHENTICATION) {
-		memset(dcid, 0, sizeof(scid) * i);
+		net_buf_add_mem(buf, dcid, sizeof(scid));
```

Comment:
This is also wrong, it should have been sizeof(scid) *i as the original code, although the memset above was probably wrong as well since it was reseting all dcids to 0 when in fact some may have been accepted.

Reply:
The problem I found is that if early error occurs (which is basically "All connections refused" case) i = 0, so no CIDs are sent back. I updated a PR